### PR TITLE
Reloading realism pass

### LINKED ISSUE #37293: M855A1 has incorrect materials when disassembled.
Game version: 0.D-11505-gfe7e3c4 (tiles) 

Operating system: Windows 10

Tiles or curses: Tiles

Mods active: 

### Expected behavior
M855A1, being a lead free round, shouldn't have lead as a component when disassembled, only copper and steel.

### Actual behavior
M855A1 yields up to 3 lead when disassembled.

### Steps to reproduce the behavior
Disassemble M855A1 at any time.

### PR DESCRIPTION
**Pull request by Rogal-Dorn**
#### Summary
SUMMARY: Balance "Bring reloading components close to reality, make brass spawn in packs, make gunpowder units 100 mg"

#### Purpose of change
Gunpowder requirements are all over the place, and I believe universally are too high. The gram is the current unit of gunpowder, which equates to about 15.4 grains. Most pistol rounds have a charge below this, and some rounds (mostly .22 LR) are below a single grain. Some notable errors were .22 LR (old value: 2 grams, new value: 65 mg, 30 times less), M576 and derivatives ( old value: 35 grams, new value: 186 mg, 188 times less) and just about every "normal" cartridge having about 4x the amount of powder they should. 
Projectile materials were generally incorrect but were pretty close in a few places. Many rounds with steel components did not take any steel to produce. Notably 7.62x39mm M43 and 5.56 M855A1 which both take scrap metal now. As a general rule lead requirements are too low and many rounds don't require copper when they should.
Fixes #37293
Brass spawns in units of 10-20 instead of large packs as is realistic.

#### Describe the solution
Changed unit of gunpowder to 100 mg
Ammunition casings now spawn in packs of 50/100/250/500 instead of 10-20, with smaller packs being much more common. Relative rates were untouched.
Gunpowder, lead, copper, and steel (scrap metal) requirements are based on the following
https://explosia.cz/app/uploads/2019/07/EXPLOSIA_reloading2019_en_new.pdf
https://www.alliantpowder.com/resources/catalog/2019_Alliant_Powder_Catalog.pdf
https://www.shootersreference.com/reloadingdata/
https://www.militarynewbie.com/wp-content/uploads/2013/11/TM-43-0001-28-Army-Ammunition-Data-Sheets.pdf
Many rounds were not available in these sources and some wheren't available in any official release at all. Some rounds are based off factory loads with unpublished components, some are based off of military cartridges in non-english speaking countries, some are fictional, and some are extremely obscure. In several cases I make guesses and give arbitrary values, but I try to avoid this whenever possible. Flechettes are the most notable place where this happens despite them being used by the American military. If you find definitive sources or anything at all really that would help improve those please share them.
I was unable to find exact proportions of copper-lead-steel in basically any projectile so as a general rule I used 80% lead-20% copper, used 1 scrap metal for rounds with steel penetrators, and left other portions (incendiary, magnesium, scrap for improvised 00) alone, and always round up. 
Black powder was left alone as I know basically nothing about it. It still has 1 gram units and original charge weights.
Uncraft recipes were updated so that they are in line with their crafted counterparts where applicable. 

#### Describe alternatives you've considered
Keeping gram scale and simply rounding tot the nearest gram (would be 1 in the case of most pistol rounds, 1 or 2 for most rifle rounds). 
Changing lead and copper units to 1 gram or 1 mg (currently 3g and 2g respectively)
Removing most of these recipes as out of reach with currently implemented tooling, adding cast lead bullet versions (intend to do this eventually, add tools needed to realistically jacket rounds, possibly realistically craft penetrators). 

#### Testing
Debug spawned materials, crafted ammo, disassembled factory ammo, tested itemgroups for bulk casings, 
image
image
image
image
image
image
image
image

#### Additional context
Please glance through the new values and tell me if anything looks insane! This is my largest PR and I have a terrible feeling I've screwed something up horribly.

### CODE REVIEW

**Comment by Amoebka** [comment]
This is rather bad from the human readability perspective. When I loot 454.000 units of gunpowder and a round requires 1.166 to craft, it becomes hard to tell how many round I can craft, how much I want to grab from my stash to bring to the crafting area, etc.

Big numbers are generally hard to work with. I think just rounding it to 1-2 units per round would be much better from the sanity perspective.

**Comment by Rogal-Dorn** [comment]
> 
> 
> This is rather bad from the human readability perspective. When I loot 454.000 units of gunpowder and a round requires 1.166 to craft, it becomes hard to tell how many round I can craft, how much I want to grab from my stash to bring to the crafting area, etc.
> 
> Big numbers are generally hard to work with. I think just rounding it to 1-2 units per round would be much better from the sanity perspective.

This is a place where having a decimal would be useful. I don't think it's possible

**Comment by Rogal-Dorn** [comment]
Everything has been divided by 100 as the new basic gunpower unit is 100 mg instead of 1 for readability. There are relatively few cases where this has a major impact as far as proportional change in powder used with .22 being the largest by a significant margin.

**Comment by tenmillimaster** [comment]
Looks good. >Many rounds with steel components did not take any steel to produce. Notably 7.62x39mm M43 and 5.56 M855A1 which both take scrap metal now. As a general rule lead requirements are too low and many rounds don't require copper when they should.

That's a step backwards, but I won't complain too much, overall.

**Comment by Rogal-Dorn** [comment]
> 
> 
> Looks good.
> 
> > Many rounds with steel components did not take any steel to produce. Notably 7.62x39mm M43 and 5.56 M855A1 which both take scrap metal now. As a general rule lead requirements are too low and many rounds don't require copper when they should.
> 
> That's a step backwards, but I won't complain too much, overall.

I believe I worded that section poorly. Prior to changes lead requirements were often too low and didn't require copper, but after this PR they should generally require more lead and copper than their bullet weight needs due to rounding up.

**Comment by tenmillimaster** [comment]
> I believe I worded that section poorly. Prior to changes lead requirements were often too low and didn't require copper, but after this PR they should generally require more lead and copper than their bullet weight needs due to rounding up.

No, the step backwards is letting them be made at all (from bare, unfinished materials that is), since that's not a very simple task.

**Comment by Rogal-Dorn** [comment]
> 
> 
> > I believe I worded that section poorly. Prior to changes lead requirements were often too low and didn't require copper, but after this PR they should generally require more lead and copper than their bullet weight needs due to rounding up.
> 
> No, the step backwards is letting them be made at all (from bare, unfinished materials that is), since that's not a very simple task.

Oh, yes, of course. Out of scope for this PR but I agree completely. The current recipes should generally be replaced with solid lead bullets with different stats and more elaborate methods and skills be needed for the current quality.

**Comment by Rail-Runner** [comment]
> Black powder was left alone as I know basically nothing about it. It still has 1 gram units and original charge weights.

According to #31553, you'd want blackpowder cartridge recipes to use 1.51 times as much propellant (by weight) than equivalent smokeless cartridges.

File: data/json/npcs/items_generic.json [hunk 1]
```diff
@@ -582,7 +582,7 @@
       [ "grip", 1 ],
       [ "gum", 8 ],
       [ "gun_crossbow", 1 ],
-      [ "gunpowder", 5 ],
+      [ "gunpowder", 5000 ],
```
Comment (Rail-Runner) [inline data/json/npcs/items_generic.json:585]: Is this necessary? I think it'd increase spawn rates for gunpowder by a lot, which likely has nothing to do with gunpowder having smaller charges now.

Comment (Rogal-Dorn) [reply data/json/npcs/items_generic.json]: It isn't at all, nice catch. I was searching in files for gunpowder and didn't check to see if that was weight or amount of items.

**Comment by kholat** [comment]
I have a problem, you say that now 1 unit = 1mg of powder, but then in 40mm grenade section you comment for instance "2 100mg pieces" and put 2 units of shotgun powder, so 1 unit of shotgun powder is in fact 100mg not 1mg now If I am not mistaken? Also then the recipe is wrong for instance for mixed gunpowder, since 1 shotgun powder should make 100 units of mixed but now 10 units makes 10 mixed like any other.

Either the amount that 1 shotgun powder makes is 1mg should actually be changed to 100mg or all the numbers where shotgun powder is used should be changed.

**Pull request merged**